.github/workflows/main.yml

```diff
@@ -0,0 +1,64 @@
+name: onetest
+
+on: [push]
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  one_test:
+    runs-on: ubuntu-latest
+    permissions:
+     contents: read
+     packages: write
+
+    steps:
+    - name: Git clone my repo
+      uses: actions/checkout@v3
+
+    - name: Clone repo
+      run: |
+        git clone https://github.com/ganatan/angular-react-starter.git  
+        
+    - name: START angular test
+      run: |
+        cd angular-react-starter
+        cd angular
+        npm install
+  
+  
+    - name: Build
+      run: |
+        cd angular-react-starter
+        cd angular
+        npm run ng build
+        cd dist     
+        
+    - name: Make images
+      run: |
+        docker build -t myang -f /home/runner/work/angular-react-starter/angular-react-starter/Dockerfile .
+    
+      
+    - name: Log in to the Container registry
```

Comment:
All steps below should depend on a custom step or job "test". I.e. if test fail - you should not push the built image to the remote docker registry, otherwise we may have a downtime deploying code with bugs inside.